libraries/core/src/main/kotlin/com/bumble/appyx/core/node/ParentNode.kt

```diff
@@ -98,15 +99,19 @@ abstract class ParentNode<NavTarget : Any>(
         navTarget: NavTarget,
         decorator: @Composable (child: ChildRenderer) -> Unit
     ) {
-        var child by remember { mutableStateOf<ChildEntry.Initialized<*>?>(null) }
         LaunchedEffect(navTarget) {
-            permanentNavModel.elements.collect { elements ->
-                val navKey = elements.find { it.key.navTarget == navTarget }?.key
-                    ?: NavKey(navTarget).also { permanentNavModel.add(it) }
-                child = childOrCreate(navKey)
-            }
+            permanentNavModel.addUnique(navTarget)
         }
-
+        val child by remember(navTarget) {
+            permanentNavModel
+                .elements
+                .map { navElements ->
```

Comment:
mapState? Also, StateFlows are already 'distinctUntilChanged', so you don't need that operator if you do this.

Reply:
`StateFlows` are, but `map` result is not.